src/macaron/slsa_analyzer/package_registry/pypi_registry.py

```diff
@@ -0,0 +1,198 @@
+        """Get upload time of the latest release.
+
+        Returns
+        -------
+            str | None: Upload time of latest release.
+        """
+        distributions = self.base_metadata.get("urls", None)
+        return distributions[0].get("upload_time") if distributions is not None else None
+
+    def get_package_page(self) -> str | None:
+        """Implement custom API to get package main page.
+
+        Returns
+        -------
+            str | None: Package main page.
+        """
+        url = os.path.join(self.base_url, "project", self.package)
+        response = send_get_http_raw(url, headers=self.headers)
+        if response:
+            html_snippets = response.content.decode("utf-8")
+            return html_snippets
+        return None
+
+    def get_maintainer_of_package(self) -> list | None:
+        """Implement custom API to get all maintainers of the package.
+
+        Returns
+        -------
+            list | None: Maintainers.
+        """
+        package_page: str | None = self.get_package_page()
+        if package_page is None:
+            return None
+        soup = BeautifulSoup(package_page, "html.parser")
+        maintainers = soup.find_all("span", class_="sidebar-section__user-gravatar-text")
+        return list({maintainer.get_text(strip=True) for maintainer in maintainers})
+
+    def get_maintainer_profile_page(self, username: str) -> str | None:
+        """Implement custom API to get maintainer's profile page.
+
+        Parameters
+        ----------
+            username (str): Maintainer's name.
+
+        Returns
+        -------
+            str | None: Profile page.
+        """
+        url = os.path.join(self.base_url, "user", username)
+        response = send_get_http_raw(url, headers=self.headers)
+        if response:
+            html_snippets = response.content.decode("utf-8")
+            return html_snippets
+        return None
+
+    def get_maintainer_join_date(self, maintainer: str) -> datetime | None:
+        """Implement custom API to get the maintainers join date.
+
+        Parameters
+        ----------
+            maintainer (_type_): Username.
+
+        Returns
+        -------
+            datetime | None: Maintainers join date. Only recent maintainer's data available.
+        """
+        user_page: str | None = self.get_maintainer_profile_page(maintainer)
+        if user_page:
+            pattern = r'Joined <time datetime="([^"]+)"'
+            match = re.search(pattern, user_page)
+            # Check if a match is found
+            if match:
+                # Extract the matched timestamp
+                datetime_str = match.group(1)
+                # Define the format of the datetime string
+                datetime_format = "%Y-%m-%dT%H:%M:%S%z"
+                # Return the parsed string to a datetime object
+                return datetime.strptime(datetime_str, datetime_format).replace(tzinfo=None)
```

Comment:
`datetime.strptime` should be checked for exception when `datetime_str` is in an invalid state.

Reply:
Fixed this in ea48367e7b773cb2ac6cfae4b7609b71706179b1

Reply:
I continue this discussion here - https://github.com/oracle/macaron/pull/750#discussion_r1642036458